blockchain/network.py

```diff
@@ -463,6 +470,75 @@ def timed_receipt_request(self):
+                subscription_id = subscription['subscription_id']
+                # convert to thrift signature for sending to server
+                subscription_signature = thrift_converter.convert_to_thrift_signature(subscription['signature'])
+                # subscription already approved, request data from server
+                if subscription['status'] == "approved":
+                    subscription_response = subscription_node.client.subscription_request(subscription_id, subscription_signature)
+                    self.insert_transactions(map(thrift_converter.convert_thrift_transaction, subscription_response.transactions))
+                    self.insert_verifications(map(thrift_converter.convert_thrift_verification, subscription_response.verification_records))
+                elif subscription['status'] == "pending":
+                    logger().warning("Subscription still in pending status... Waiting for admin(s) approval.")
+
+    def get_subscription_node(self, subscription):
+        """ check if client is connected to node subscribed to and return that node. if not, attempt to connect to it and return. """
+        if not self.subscription_connected(subscription):
+            self.connect_subscription_node(subscription)
+        for node in self.connections:
+            if subscription["subscribed_node_id"] == node.node_id:
+                return node
+        return None
+
+    def subscription_connected(self, subscription):
+        """ check if connected to subscription node """
+        for node in self.connections:
+            if subscription["subscribed_node_id"] == node.node_id:
+                return True
+        return False
+
+    def connect_subscription_node(self, subscription):
+        """ connect to subscription node """
+        # check if node is already in database and just not connected
+        node = net_dao.get(subscription["subscribed_node_id"])
+        if node:
+            subscription_node = Node(node["node_id"], node["node_owner"], node["host"], node["port"], node["phases"])
+            if node["phases"] not in self.peer_dict:
+                self.peer_dict.setdefault(node["phases"], [])
+            try:
+                self.connect_node(subscription_node, node["phases"])
+            except:
+                logger().warning("Failed to connect to subscription node %s", node['node_id'])
+        else:
+            # insert new node into table and recursively call connect_subscription_node
+            node = Node(subscription['subscribed_node_id'], subscription['node_owner'], subscription['host'], subscription['port'], "00001")
+            net_dao.insert_node(node)
+            self.connect_subscription_node(subscription)
+
+    def insert_transactions(self, transactions):
+        """ insert given transactions into database """
+        for txn in transactions:
+            try:
+                transaction_db.insert_transaction(txn)
```

Comment:
What if I have a smart contract that I want to execute on transactions I've subscribed to? Let's talk to @j0j0r0 about this, but it may be another place to trigger a transaction event.